production/helm/loki/templates/backend/statefulset-backend.yaml

```diff
@@ -267,6 +267,12 @@ spec:
       kind: PersistentVolumeClaim
       metadata:
         name: data
+        {{- if .Values.backend.persistence.annotations }}
```

Comment:
We wrap the `with` in an `if` for backwards compatibility, to avoid making an unnecessary changes to immutable fields of the `StatefulSets` for existing `loki` installations.

Reply:
why would just using a `with` cause changes?

Reply:
https://helm.sh/docs/chart_template_guide/control_structures/#modifying-scope-using-with

> Note that we removed the if conditional from the previous exercise because it is now unnecessary - the block after with only executes if the value of PIPELINE is not empty.

Right sorry for some reason I think it'd do:
```
annotations:
  {}
```
w/o the `if` and make a "breaking" change to the STS' volumeClaimTemplate. Confirmed that's not the case so will revert as part resolving conflicts.